packages/trace-viewer/src/ui/timeline.tsx

```diff
@@ -282,7 +284,10 @@ function calculateDividerOffsets(clientWidth: number, boundaries: Boundaries): {
   const offsets = [];
   for (let i = 0; i < dividerCount; ++i) {
     const time = firstDividerTime + sectionTime * i;
-    offsets.push({ position: timeToPosition(clientWidth, boundaries, time), time });
+    const position = timeToPosition(clientWidth, boundaries, time);
+    if ((position + 1 /* timeline-divider width */) > clientWidth)
```

Comment:
IIUC, this is fighting the scroll bar that appears? Perhaps you can avoid all of this by setting `overflow: clip` or `overflow: hidden` on some timeline container?